client/app/admin/classcontrol/classcontrol.controller.js

```diff
@@ -48,6 +49,54 @@ angular.module('observatory3App')
         });
     };
 
+    var getSemesterToday = function(){
+      var today = new Date();
+      var semester = ""
+      if (today.getMonth() < 5){
+        semester = "spring"
+      }
+      else if (today.getMonth() < 8){
+        semester = "summer"
+      }
+      else{
+        semester = "fall"
+      }
+      return {year: String(today.getFullYear()),
+              semester: semester};
+    }
+
+    $scope.createNewSemester = function(form){
+      $scope.submitted = true;
+      var thisSemester = "";
+      if ($scope.newSemester.year !== ""){
+        thisSemester += $scope.newSemester.year;
+      }
+      else{
+        thisSemester += $scope.todaySemester.year;
+      }
+      if ($scope.newSemester.semester !== ""){
+        thisSemester += $scope.newSemester.semester;
+      }
+      else{
+        thisSemester += $scope.todaySemester.semester;
+      }
+      if(form.$valid) {
+        $http.post("/api/classyear/", {
+          semester: thisSemester
+        }).success(function(){
+            updateClassYear();
+            $scope.newSemester = {year:"", semester:""};
+            $scope.showNewSemester = false;
+          })
+          .error(function(err){
+            console.error("Error creating new semester", err);
```

Comment:
i think we should silence this for now but in the future log js errors to the server